core/templates/dev/head/app.js

```diff
@@ -30,9 +30,10 @@ var oppia = angular.module(
     window.GLOBALS ? (window.GLOBALS.ADDITIONAL_ANGULAR_MODULES || [])
                    : []));
 
-// TODO(sll): Get this to read from a common JSON file; it's replicated in
-// feconf.
-oppia.constant('CATEGORY_LIST', GLOBALS.ALL_CATEGORIES);
+for (var constantName in CONSTANTS) {
+  oppia.constant(constantName, CONSTANTS[constantName]);
```

Comment:
Could you suggest how would you setup the fake `constants`? Because I have tried setting up constants variable in `beforeEach` but it didn't work and the `constants` variable still contains values defined in *constants.json*. Only way that I was able to partially test this three lines, was to wrap them in function and use this code
```javascript
describe('Constants Generating', function() {
  beforeEach(module('oppia'));

  var tc = null;
  beforeEach(function () {
    tc = window.transformConstants;
  });

  it('should transform key value pairs to angular constants', function() {
    constants = {'CONSTANT1': [1, 2, 3],
                 'CONSTANT2': 'something',
                 'CONSTANT3': 5};
    tc(constants);
    inject(function($injector) {
      expect($injector.has('CONSTANT1')).toBeTruthy();
      expect($injector.get('CONSTANT1')).toBe(constants.CONSTANT1);
      expect($injector.has('CONSTANT2')).toBeTruthy();
      expect($injector.get('CONSTANT2')).toBe(constants.CONSTANT2);
      expect($injector.has('CONSTANT3')).toBeTruthy();
      expect($injector.get('CONSTANT3')).toBe(constants.CONSTANT3);
      expect($injector.has('NOT_EXISTING')).toBeFalsy();
    });
  });
});
```
but it seems pretty dirty for me.